core/includes/classes/class-wp-rag-page-ai-configuration.php

```diff
@@ -213,7 +278,7 @@ public function number_of_documents_field_render() {
 		?>
 		<input type="number" name="<?php echo self::OPTION_NAME; ?>[search][number_of_documents]"
 				value="<?php echo esc_attr( $options['search']['number_of_documents'] ?? '' ); ?>"
-				min="1" max="8"
+				min="1" max="8" required
```

Comment:
It would be better for the user to be able to leave it blank, in which case, the default value would be used on the API side. Alternatively, the form could set the default value when the value isn't set yet, but we would have to sync the default value of the API and plugin.

Also, this change has nothing to do with the model config feature, and should be in a separate PR.